pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -672,14 +683,19 @@ func translateHeaderMatch(name string, in *networking.StringMatch, node *model.P
 		// Golang has a slightly different regex grammar
 		out.HeaderMatchSpecifier = &route.HeaderMatcher_PrefixMatch{PrefixMatch: m.Prefix}
 	case *networking.StringMatch_Regex:
-		if !util.IsIstioVersionGE14(node) {
-			out.HeaderMatchSpecifier = &route.HeaderMatcher_RegexMatch{RegexMatch: m.Regex}
-		} else {
-			out.HeaderMatchSpecifier = &route.HeaderMatcher_SafeRegexMatch{
-				SafeRegexMatch: &matcher.RegexMatcher{
-					EngineType: regexEngine,
-					Regex:      m.Regex,
-				},
+		switch m.Regex {
+		case ".*", "^.*", ".*$", "^.*$":
+			out.HeaderMatchSpecifier = &route.HeaderMatcher_PresentMatch{PresentMatch: true}
+		default:
+			if !util.IsIstioVersionGE14(node) {
```

Comment:
since youare touching this, nuke this if block. default to safe regex always as its there in 1.4+ istio-proxies

Reply:
Then do you want me to modify this one to use safe regex always if 1.4+ istio-proxies are used?

Reply:
Please check out my revised code, once again. Thanks.